app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,52 @@
+class MerchantsController < ApplicationController
+  def index
+
+  end
+
+  def show
+    @merchant = Merchant.find(params[:id])
+    @merchant_products = @merchant.products
```

Comment:
Since the `@merchant` variable contains the products that you need access to - I would err on the side of not including this extra instance variable since you can just use `@merchant` to gain access to the product information